scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -1026,15 +1042,43 @@ public List<FilteredTopWorkflow> getTopWorkflowsWithIssues(int numberOfWorkflows
         });
     }
 
+    public Map<String, Integer> getNumberOfJobsSubmittedFromEachPortals(final String workflowName, String bucketName,
+            String user, String tenant, final long startTime, final long endTime) {
+
+        return executeReadOnlyTransaction(session -> {
+
+            String selectSubQuery = "select submissionMode, count(*) as numberOfJobs FROM JobData where startTime > 0 and finishedTime > 0  ";
+            String subQueryGroupByStatement = "group by submissionMode order by numberOfJobs desc";
+            Query query = getTopWorkflowsQuery(session,
+                                               workflowName,
+                                               bucketName,
+                                               user,
+                                               tenant,
+                                               startTime,
+                                               endTime,
```

Comment:
startTime and endTime are using finishedTime column inside this method, consider adding an extra parameter to this method to allow switching between finishedTime, submittedTime, startTime, ...